mesoscope/cellpack
# Automated cleanup of expired result metadata in staging firebase database

Problem
=======
What is the problem this work solves, including
closes #195 

Solution
========
What I/we did to solve this problem
- scheduled a github action to weekly cleanup results metadata in the staging firebase database 
- added github secrets for the staging db credentials to our repo, named `FIREBASE_TOKEN` and `FIREBASE_EMAIL`  
- implemented the logic in `FirebaseHandler` and `DBRecipeHandler` to verify the existence of packing results in the S3 bucket, if not found, remove the corresponding metadata from firebase


## Type of change
Please delete options that are not relevant.
* New feature (non-breaking change which adds functionality)


Steps to Verify:
----------------
1. View the most recent cleanup github action run for testing at this [link](https://github.com/mesoscope/cellpack/actions/runs/8440729655/job/23118306690)

Or test the main functions locally:
1. pack a test recipe, and run `python cellpack/bin/cleanup_tasks.py`. Temporarily adjust the condition from `if result_age.days > 180 and not self.validate_existence(result_data["url"]):` to something like `if result_age.total_seconds() < 1000`. To check if the metadata of your test recipe gets deleted as expected
